src/services/contract-service.ts

```diff
@@ -579,51 +544,51 @@ export class ContractService implements ContractReads, ContractWrites {
         throw new Error(enhancedMessage);
       }
 
-      // For other gas estimation failures, use fallback
+      // Handle user rejection specifically
+      if (errorMessage.includes("rejected") || errorMessage.includes("denied") || errorMessage.includes("user rejected")) {
+        throw new Error("Redeem transaction was cancelled by user");
+      }
 
-      gasEstimate = 400000n;
```

Comment:
I think it might have been tested for the specific smart contract method before setting here.